### ISSUE TITLE
Find permanent solution for handling unknown tasks/projects

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#903]
Ref #901

**Comment by simonrozsival** [comment]
Is there any progress in discussions with the backend team, @amulware?

**Comment by amulware** [comment]
Ths solution has been decided on. I'll still work out some more details, but briefly: we will implement a proper full-sync that downloads all user data and then perform a local comparison to handle local unsynced data. This will not run on every sync though, but maybe only once a day to once a week or so.

While this is sort of brute forcing it, we will need this logic in the long-term anyways, to handle users that haven't been to the app in weeks/months since we cannot use the ?since endpoints in that case either.

Implementing this is not a priority for the public release however, it is fine if we only ship it 1-2 weeks after.

In the meantime, backend is working on creating endpoints that we can use to get removed-from/deleted/archived entities using a ?since parameter with. In the long-term we will switch to those for the most part, and only use the above solution when the user has been gone for more than a few weeks.

Details to be determined.

**Comment by simonrozsival** [comment]
By "proper full-sync" you mean omitting the `?since=` parameter and fetching "all" data? At least for entities other than time entries that makes sense.

We will then diff local and the remote entities to see which were archived?

**Comment by amulware** [comment]
Yeah, something like that. It'll be more complicated because if we lose access to a workspace everything in that workspace has to be gone, but if we have unsynced data... well :D

**Comment by amulware** [comment]
I just realised that this will not help us with the issue that some time entries will refer to archived tasks/projects. For now we can keep our hack, but I'll see what to do about this down the line as well.
